pallets/baby-liminal/src/lib.rs

```diff
@@ -334,71 +342,60 @@ pub mod pallet {
-                    Some(T::WeightInfo::verify_data_deserializing_fails(proof_len)),
-                )
-            })?;
+            let proof: Proof<Curve> = CanonicalDeserialize::deserialize_compressed(&*proof)
+                .map_err(|e| {
+                    log::error!("Deserializing proof failed: {e:?}");
+                    (
+                        Error::<T>::DeserializingProofFailed,
+                        Some(Weight::default()),
+                    )
+                })?;
 
-            let public_input: Vec<S::CircuitField> =
-                CanonicalDeserialize::deserialize(&*public_input).map_err(|e| {
-                    log::error!("Deserializing public input failed: {:?}", e);
+            let public_input: Vec<CircuitField> =
+                CanonicalDeserialize::deserialize_compressed(&*public_input).map_err(|e| {
+                    log::error!("Deserializing public input failed: {e:?}");
                     (
                         Error::<T>::DeserializingPublicInputFailed,
-                        Some(T::WeightInfo::verify_data_deserializing_fails(
-                            proof_len + public_input.len() as u32,
-                        )),
+                        Some(Weight::default()),
                     )
                 })?;
 
             let verification_key =
                 VerificationKeys::<T>::get(verification_key_identifier).ok_or((
                     Error::<T>::UnknownVerificationKeyIdentifier,
-                    Some(T::WeightInfo::verify_key_deserializing_fails(0)),
+                    Some(Weight::default()),
                 ))?;
-            let verification_key: S::VerifyingKey =
-                CanonicalDeserialize::deserialize(&**verification_key).map_err(|e| {
+            let verification_key: VerifyingKey<Curve> =
+                CanonicalDeserialize::deserialize_compressed(&**verification_key).map_err(|e| {
                     log::error!("Deserializing verification key failed: {:?}", e);
                     (
                         Error::<T>::DeserializingVerificationKeyFailed,
-                        Some(T::WeightInfo::verify_key_deserializing_fails(
-                            verification_key.len() as u32,
-                        )),
+                        Some(Weight::default()),
                     )
                 })?;
 
-            let valid_proof = S::verify(&verification_key, &public_input, &proof)
-                .map_err(|err| (Error::<T>::VerificationFailed(err), None))?;
-
-            ensure!(valid_proof, (Error::<T>::IncorrectProof, None));
-
-            Self::deposit_event(Event::VerificationSucceeded);
-            Ok(())
+            match PlonkKzgSnark::verify::<StandardTranscript>(
+                &verification_key,
+                &public_input,
+                &proof,
+                None,
+            ) {
+                Ok(_) => {
+                    Self::deposit_event(Event::VerificationSucceeded);
```

Comment:
perhaps this event could carry more info: proof and input or v-key identifier at the very least?

Reply:
both proof and input might be pretty big (kilobytes), so I just included the v-key identifier